### PR TITLE
Update CONTRIBUTING.md guide

### PR DESCRIPTION
**Pull request by alexellis**
For attention @johnmccabe @rgee0 - any additional points we wanted to cover/explain?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,15 +2,35 @@
...
+* A Pull Request is not necessary. Raise an [Issue](https://github.com/alexellis/faas/issues) and we'll fix it as soon as we can. 
 
-* Please always provide a summary of what you changed, how you did it and how it can be tested by project maintainers. 
+**I have a [great] idea**
+
+The OpenFaaS maintainers would like to make OpenFaaS the best it can be and we can't do it alone. Our time is limited so we'd like to make sure we agree on the proposed work before you spend time doing it. Saying "no" is hard which is why we'd rather say "yes" ahead of time.
+
+What makes a good proposal?
+
+* A quick summary
```
Comment (rgee0) [inline CONTRIBUTING.md]: Including motivation?

**Comment by rgee0** [comment]
Looks good, provides an extra level of detail and explains why.  Left one comment about understanding their motivation as that might give us some additional insight when evaluating.

File: CONTRIBUTING.md (hunk 1 above)
Comment (johnmccabe) [reply CONTRIBUTING.md]: pedantic hat on but `short` rather than `quick`

Comment (alexellis) [reply CONTRIBUTING.md]: As a bullet OK?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,15 +2,35 @@
...
+* Pros + Cons
+* Effort required
+* Screenshots or CLI examples
```
Comment (johnmccabe) [inline CONTRIBUTING.md]: Mock screenshots..

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,15 +2,35 @@
...
+
+**Paperwork for Pull Requests**
+
+Please read this whole guide and make sure you agree to our DCO agreement (included below):
+
+* Sign-off your commits
+* Use the whole template for issues and pull requests
```
Comment (johnmccabe) [inline CONTRIBUTING.md]: Complete the whole template

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,15 +2,35 @@
...
+* A quick summary
+* Pros + Cons
+* Effort required
+* Screenshots or CLI examples
+
+**Paperwork for Pull Requests**
+
+Please read this whole guide and make sure you agree to our DCO agreement (included below):
+
+* Sign-off your commits
```
Comment (johnmccabe) [inline CONTRIBUTING.md]: Link to #sign-your-work

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -2,15 +2,35 @@
...
+* Use the whole template for issues and pull requests
+* Always give instructions for testing
+ * Give us CLI commands and output or screenshots where you can 
+
+**Questions/suggestions/support needed**
```
Comment (johnmccabe) [inline CONTRIBUTING.md]: I have a question, a suggestion or need help - perhaps - torn on this suggestion tho

File: CONTRIBUTING.md (hunk 4 above)
Comment (johnmccabe) [reply CONTRIBUTING.md]: Do we actually link to the DCO anywhere? https://developercertificate.org/

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -2,15 +2,35 @@
...
+**I've found a typo**
 
-* Don't raise PRs for typos, these aren't necessary - just raise an Issue
-* If the documentation can be improved / translated etc please raise an issue to discuss. 
+* A Pull Request is not necessary. Raise an [Issue](https://github.com/alexellis/faas/issues) and we'll fix it as soon as we can. 
 
-* Please always provide a summary of what you changed, how you did it and how it can be tested by project maintainers. 
+**I have a [great] idea**
+
+The OpenFaaS maintainers would like to make OpenFaaS the best it can be and we can't do it alone. Our time is limited so we'd like to make sure we agree on the proposed work before you spend time doing it. Saying "no" is hard which is why we'd rather say "yes" ahead of time.
```
Comment (johnmccabe) [inline CONTRIBUTING.md]: "can't do it alone" - "can't do it without your help"? A bit friendlier perhaps?

File: CONTRIBUTING.md (hunk 4 above)
Comment (alexellis) [reply CONTRIBUTING.md]: The body is included in the guide.

**Comment by alexellis** [comment]
Feedback added.

**Pull request merged**